### ISSUE TITLE
ML: build failure when using C89

### ISSUE THREAD

**Issue by sethrj** [trilinos/Trilinos#6667]
## Bug Report
@trilinos/ml

### Description

Trying to build the ML package on Windows Intel 15, it appears the default behavior is to compile with strict C89. There is a single declaration in ML's C code that's noncompliant.

```
--- a/packages/ml/src/Smoother/ml_smoother.c
+++ b/packages/ml/src/Smoother/ml_smoother.c
@@ -7367,10 +7367,11 @@ int ML_Smoother_HiptmairSubsmoother_Create(ML **ml_subpr
 #ifdef BLOCKMLS
   int nblocks, *block_list;
 #endif
+   ML_Comm *comm;

    ML_Create(ml_subproblem,1);
    /*Replace the default MPI_COMM_WORLD inside ml_subproblem's communicator.*/
-   ML_Comm *comm = (*ml_subproblem)->comm;
+   comm = (*ml_subproblem)->comm;
    comm->USR_comm = Amat->comm->USR_comm;
    comm->ML_mypid = Amat->comm->ML_mypid;
    comm->ML_nprocs = Amat->comm->ML_nprocs;
```

### Steps to Reproduce

Enable ML, build with Intel 2015 on Windows.

```
"C:/Program Files/CMake/bin/ctest.exe" --launch --target-name ml --build-dir F:\s3j\scale\build\Trilinos\packages\ml\src --output Trilinos\packages\ml\src\CMakeFiles\ml.dir\Smoother\ml_smoother.c.obj --source ..\Trilinos\packages\ml\src\Smoother\ml_smoother.c --language C --filter-prefix "Note: including file: " -- C:\PROGRA~2\Intel\COMPOS~1\bin\intel64\icl.exe  /nologo -D_CRT_NONSTDC_NO_DEPRECATE -D_CRT_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -I. -I..\packages -Ipackages -I..\Exnihilo\packages -IExnihilo\packages -I..\TriBITS\tribits\win_interface\include -ITrilinos\packages\ml\src -I..\Trilinos\packages\ml\src\Include -I..\Trilinos\packages\ml\src\Comm -I..\Trilinos\packages\ml\src\Coarsen -I..\Trilinos\packages\ml\src\FEGrid -I..\Trilinos\packages\ml\src\Main -I..\Trilinos\packages\ml\src\Smoother -I..\Trilinos\packages\ml\src\Krylov -I..\Trilinos\packages\ml\src\Operator -I..\Trilinos\packages\ml\src\Utils -I..\Trilinos\packages\ml\src\RefMaxwell -I..\Trilinos\packages\ml\src\MatrixFree -I..\Trilinos\packages\ml\src\MLAPI -I..\Trilinos\packages\ml\src\LevelWrap -ITrilinos\packages\ifpack\src -I..\Trilinos\packages\ifpack\src -I..\common\auxiliarySoftware\SuiteSparse\src\AMD\Include -I..\common\auxiliarySoftware\SuiteSparse\src\UFconfig -ITrilinos\packages\amesos\src -I..\Trilinos\packages\amesos\src -I..\Trilinos\packages\amesos\src\SuiteSparse\AMD\Include -I..\Trilinos\packages\amesos\src\SuiteSparse\COLAMD\Include -I..\Trilinos\packages\amesos\src\SuiteSparse\UFconfig -I..\Trilinos\packages\amesos\src\SuiteSparse\KLU\Include -I..\Trilinos\packages\amesos\src\SuiteSparse\BTF\Include -I..\Trilinos\packages\amesos\src\SuiteSparse\CAMD\Include -I..\Trilinos\packages\amesos\src\SuiteSparse\CCOLAMD\Include -ITrilinos\packages\epetraext\src -I..\Trilinos\packages\epetraext\src -I..\Trilinos\packages\epetraext\src\transform -I..\Trilinos\packages\epetraext\src\inout -I..\Trilinos\packages\epetraext\src\coloring -I..\Trilinos\packages\epetraext\src\model_evaluator -I..\Trilinos\packages\epetraext\src\block -I..\Trilinos\packages\epetraext\src\restrict -ITrilinos\packages\triutils\src -I..\Trilinos\packages\triutils\src -ITrilinos -ITrilinos\packages\epetra\src -I..\Trilinos\packages\epetra\src -I..\Trilinos\packages\teuchos\remainder\src -ITrilinos\packages\teuchos\remainder\src -ITrilinos\packages\teuchos\core\src -I..\Trilinos\packages\teuchos\core\src -IC:\vendors\msvc-2013\qt\4.8.6\include\QtDesigner -IC:\vendors\msvc-2013\qt\4.8.6\include\QtDeclarative -IC:\vendors\msvc-2013\qt\4.8.6\include\QtScriptTools -IC:\vendors\msvc-2013\qt\4.8.6\include\QtXml -IC:\vendors\msvc-2013\qt\4.8.6\include\QtSql -IC:\vendors\msvc-2013\qt\4.8.6\include\QtOpenGL -IC:\vendors\msvc-2013\qt\4.8.6\include\QtMultimedia -IC:\vendors\msvc-2013\qt\4.8.6\include\QtNetwork -IC:\vendors\msvc-2013\qt\4.8.6\include\phonon -IC:\vendors\msvc-2013\qt\4.8.6\include\QtXmlPatterns -IC:\vendors\msvc-2013\qt\4.8.6\include\QtHelp -IC:\vendors\msvc-2013\qt\4.8.6\include\QtUiTools -IC:\vendors\msvc-2013\qt\4.8.6\include\QtTest -IC:\vendors\msvc-2013\qt\4.8.6\include\QtScript -IC:\vendors\msvc-2013\qt\4.8.6\include\QtSvg -IC:\vendors\msvc-2013\qt\4.8.6\include\Qt3Support -IC:\vendors\msvc-2013\qt\4.8.6\include\QtGui -IC:\vendors\msvc-2013\qt\4.8.6\include\QtCore -IC:\vendors\msvc-2013\qt\4.8.6\mkspecs\default -IC:\vendors\msvc-2013\qt\4.8.6\include -I..\Trilinos\packages\teuchos\numerics\src -I..\Trilinos\packages\teuchos\comm\src -I..\Trilinos\packages\teuchos\parameterlist\src -I..\Trilinos\packages\common\auxiliarySoftware\SuiteSparse\src\AMD\Include -I..\Trilinos\packages\common\auxiliarySoftware\SuiteSparse\src\BTF\Include -I..\Trilinos\packages\common\auxiliarySoftware\SuiteSparse\src\CAMD\Include -I..\Trilinos\packages\common\auxiliarySoftware\SuiteSparse\src\CCOLAMD\Include -I..\Trilinos\packages\common\auxiliarySoftware\SuiteSparse\src\COLAMD\Include -I..\Trilinos\packages\common\auxiliarySoftware\SuiteSparse\src\KLU\Include -I..\Trilinos\packages\common\auxiliarySoftware\SuiteSparse\src\UFconfig -ITrilinos\packages\aztecoo\src -I..\Trilinos\packages\aztecoo\src /DWIN32 /D_WINDOWS /W1 /MDd /Zi /Ob0 /Od /RTC1 /showIncludes /FoTrilinos\packages\ml\src\CMakeFiles\ml.dir\Smoother\ml_smoother.c.obj /FdTrilinos\packages\ml\src\CMakeFiles\ml.dir\ml.pdb -c ..\Trilinos\packages\ml\src\Smoother\ml_smoother.c
..\Trilinos\packages\ml\src\Smoother\ml_smoother.c(7373): error: declaration may not appear after executable statement in block
     ML_Comm *comm = (*ml_subproblem)->comm;
```

**Comment by csiefer2** [comment]
Approved the PR and set it to automerge.

Thanks for the paych!

**Comment by sethrj** [comment]
Fixed by #6668
